### DISCUSSION TITLE
unresolved external symbol _guard_check_icall_$fo$

### DISCUSSION THREAD

**Discussion by htartisan** [pocoproject/poco#4694] [Q&A]
**Describe the bug**
While trying to build the "Foundation" libraries (all) using the MS Visual Studio 2019 compiler, I encountered the following error for 4 of the library versions:

libucrtd.lib(checkcfg.obj) : error LNK2001: unresolved external symbol _guard_check_icall_$fo$
bin64\static_mt\TestAppd.exe : fatal error LNK1120: 1 unresolved externals

**To Reproduce**
Copy the latest Poco libraries (all) to a source folder.
Go the the "CppUint" folder and load the "CppUnit_vs160.sln" solution file with MS Visual Studio 2019.
Go to the Build->Batch Build dialog, select all output targets, and press the "build" button.
Go the the "Foundation" folder and load the "Foundation_vs160.sln" solution file with MS Visual Studio 2019.
Go to the Build->Batch Build dialog, select all output targets, and press the "build" button.

**Expected behavior**
All output binaries should build without errors.

**Logs**
------ Rebuild All started: Project: TestApp, Configuration: debug_static_mt x64 ------
TestApp.cpp
libucrtd.lib(checkcfg.obj) : error LNK2001: unresolved external symbol _guard_check_icall_$fo$
bin64\static_mt\TestAppd.exe : fatal error LNK1120: 1 unresolved externals
Done building project "TestApp_vs160.vcxproj" -- FAILED.

**Screenshots**

**Please add relevant environment information:**
 - OS Type and Version: Windows 11
 - POCO Version: 1.13.3
 - Third-party product (eg. database or library) type and version

**Additional context**

**Comment by aleks-f** [comment]
if you want this fixed, send a pull request

**Comment by yanzhuo0323** [comment]
Are you using windows sdk 26100?

**Reply by FuXiii** [reply]
Yes, I use 26100, and the same error show up when I compile `PhysX`.

<img width="507" height="59" alt="image" src="https://github.com/user-attachments/assets/4b56888a-f7ca-49d1-b03c-6d7228b07458" />

**Comment by wddiddddddd** [comment]
VS2019(V142) use SDK 22621

**Comment by FCode-hub** [comment]
> VS2019(V142) use SDK 22621

This one definitely worked for me.

**Comment by aleks-f** [comment]
this has to do with [SDK version](https://forum.gamemaker.io/index.php?threads/solution-yyc-build-linker-error-unresolved-external-symbol-_guard_check_icall_-fo.115387/)

> Uninstall latest Windows SDK 10.0.26100, and use the previous one (10.0.22621)
> This linker issue is fixed.
> Windows SDK 10.0.26100 is the culprit of this linker error.

**Reply by esmirno** [reply]
correct SDK : 26100 has same linker error on my side, while 10.0.19041.0 worked fine. Do you know how to select particulat toolset in cmake maybe? currently i'm using  

```bash
cmake -G "Visual Studio 16 2019" -T v142 ..
```
while this selected lates version from system. So I've manually changed SDK version by opening created project in visual studio.
